### ISSUE TITLE
Add type hints or stubs

### ISSUE THREAD

**Issue by MicaelJarniac** [Rockhopper-Technologies/enlighten#54]
For static type checking with tools like Mypy.

**Comment by avylove** [comment]
It's been on my list for a while. Will need to be in pyi files. I'll try to find some time to work on it, but also happy to take PRs.

**Comment by MicaelJarniac** [comment]
It has to be on `.pyi` files because of backward compatibility with ancient Python versions, right?

**Comment by avylove** [comment]
Yes. We'll probably be supporting Python 2.7 for another 16 months until the end of standard maintenance for RHEL 7.
